platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/impl/PbjConverter.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.service.impl;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+import com.hedera.hapi.node.base.Timestamp;
+import com.hedera.hapi.platform.state.PlatformState;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.platform.NodeId;
+import com.swirlds.platform.consensus.ConsensusSnapshot;
+import com.swirlds.platform.crypto.SerializableX509Certificate;
+import com.swirlds.platform.state.MinimumJudgeInfo;
+import com.swirlds.platform.state.PlatformStateAccessor;
+import com.swirlds.platform.system.SoftwareVersion;
+import com.swirlds.platform.system.address.Address;
+import com.swirlds.platform.system.address.AddressBook;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class handles conversion from PBJ objects related to the platform state to the corresponding Java objects, and vice versa.
+ */
+public class PbjConverter {
```

Comment:
I'm just wondering why is this class necessary? If the PlatformState is modeled in protobuf, then PBJ already generates the codecs. Why do we need this converter?

Reply:
This class can be avoided but in this case I'd have to migrate multiple domain objects to PBJ. That would significantly extend the scope of this PR. Besides, I'm not sure if  that's a direction we're taking. That is, is the goal to get rid of all the domain object if favor of the objects defined in `hedera-protobuf` library?

Reply:
Consistency may be one argument in favor of having everything in protobuf. But I see your reasoning regarding the size of the PR.